site-src/geps/gep-851.md

```diff
@@ -0,0 +1,88 @@
+# GEP-851: Allow Multiple Certificate Refs per Gateway Listener
+
+* Issue: [#851](https://github.com/kubernetes-sigs/gateway-api/issues/851)
+* Status: Implementable
+
+## TLDR
+
+Replace `CertificateRef` field with an `CertificateRefs` field in Gateway
+Listeners.
+
+## Goals
+
+* Allow certs to be attached for multiple unique Hostnames for the same
+  Listener. This is most helpful when the hostname is empty on the Listener and
+  attached Routes specify multiple hostnames.
```

Comment:
TLS cert per hostname for the same port (listener collapsing) is another alternative.

The feature is helpful only in the case when you don't have a hostname present at all.
We should note that approach here in this GEP.

Reply:
I think it's also helpful when you are trying to support multiple layers of subdomains `*.example.net` on listener, `app.staging.example.net` on route, I can clarify that.

Reply:
Added a better example here. Agree that this is still possible by creating lots of listeners, but I'm not sure that's practical at a certain scale, especially with the limit on the number of listeners we have.